tests/functionaltests/Tests/FileActivationTests.mm

```diff
@@ -0,0 +1,194 @@
+
+#include <UWP/WindowsApplicationModelActivation.h>
+#include <UWP/WindowsStorage.h>
+#include <UWP/WindowsUIXaml.h>
+#include "StringHelpers.h"
+#include "UIKit/UIApplication.h"
+
+using namespace ABI::Windows::ApplicationModel::Activation;
+using namespace ABI::Windows::Foundation;
+using namespace ABI::Windows::Storage;
+using namespace ABI::Windows::Foundation::Collections;
+using namespace Microsoft::WRL;
+
+// Method to call in tests to activate app
+extern "C" void UIApplicationActivationTest(IInspectable* args, void* delegateClassName);
+
+// This is not a complete list of interfaces implemented by the real FileActivatedEventArgs but should be a
+// sufficient stand in for how it will be used.
+MOCK_CLASS(MockFileActivatedEventArgs,
+           public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IFileActivatedEventArgs, IActivatedEventArgs> {
+
+               // Claim to be the implementation for the real system RuntimeClass for FileActivatedEventArgs.
+               InspectableClass(RuntimeClass_Windows_ApplicationModel_Activation_FileActivatedEventArgs, BaseTrust);
+
+           public:
+               // IActivatedEventArgs
+               MOCK_STDCALL_METHOD_1(get_Kind);
+               MOCK_STDCALL_METHOD_1(get_PreviousExecutionState);
+               MOCK_STDCALL_METHOD_1(get_SplashScreen);
+
+               // IFileActivatedEventArgs
+               MOCK_STDCALL_METHOD_1(get_Files);
+               MOCK_STDCALL_METHOD_1(get_Verb);
+           });
+
+MOCK_CLASS(MockFileVectorView,
+           public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IVectorView<IStorageItem*>> {
+
+               InspectableClass(L"MockVectorView", BaseTrust);
+
+               MOCK_STDCALL_METHOD_2(GetAt);
+               MOCK_STDCALL_METHOD_1(get_Size);
+               MOCK_STDCALL_METHOD_3(IndexOf);
+           });
+
+MOCK_CLASS(MockStorageItem,
+           public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IStorageItem> {
+
+               InspectableClass(RuntimeClass_Windows_Storage_StorageFile, BaseTrust);
+
+               MOCK_STDCALL_METHOD_2(RenameAsyncOverloadDefaultOptions);
+               MOCK_STDCALL_METHOD_3(RenameAsync);
+               MOCK_STDCALL_METHOD_1(DeleteAsyncOverloadDefaultOptions);
+               MOCK_STDCALL_METHOD_2(DeleteAsync);
+               MOCK_STDCALL_METHOD_1(GetBasicPropertiesAsync);
+               MOCK_STDCALL_METHOD_1(get_Name);
+               MOCK_STDCALL_METHOD_1(get_Path);
+               MOCK_STDCALL_METHOD_1(get_Attributes);
+               MOCK_STDCALL_METHOD_1(get_DateCreated);
+               MOCK_STDCALL_METHOD_2(IsOfType);
+           });
+
+// Delegate for testing File foreground activation
+@interface FileActivationForegroundTestDelegate : NSObject <UIApplicationDelegate>
+@property (nonatomic, readonly) StrongId<NSMutableDictionary> methodsCalled;
+@end
+
+@implementation FileActivationForegroundTestDelegate
+- (id)init {
+    self = [super init];
```

Comment:
I'll update yours too I guess

---
In reply to: [82295491](https://github.com/Microsoft/WinObjC/pull/1100#discussion_r82295491) [](ancestors = 82295491)